src/main/kotlin/dev/arbjerg/ukulele/features/LeaveOnIdleService.kt

```diff
@@ -15,34 +15,23 @@ class LeaveOnIdleService(private val scheduler: TaskScheduler, private val botPr
     private val log: Logger = LoggerFactory.getLogger(LeaveOnIdleService::class.java)
     private val timers: ConcurrentHashMap<Long, ScheduledFuture<*>> = ConcurrentHashMap()
 
-    /**
-     * Runnable that gets passed to the Spring TaskScheduler. Used to keep track of how many minutes
-     * a player in a guild has spent idling.
-     */
-    class IdleTask(val guild: Guild, private val idleTimeMinutes: Int) : Runnable {
-        private var numberOfMinutesIdleElapsed = -1
-
+    inner class IdleTask(val guild: Guild) : Runnable {
```

Comment:
At this point when it's just a single statement I think you should replace this with an inline lambda. E.g.

```kotlin
scheduler.schedule(instant) {
    guild.audioManager.closeAudioConnection()
}
```

(Not entirely sure if this is valid syntax)